src/libraries/System.Text.Json/tests/Common/ReferenceHandlerTests/ReferenceHandlerTests.Deserialize.cs

```diff
@@ -690,7 +690,7 @@ public async Task ObjectNull()
             }";
 
             JsonException ex = await Assert.ThrowsAsync<JsonException>(async () => await Serializer.DeserializeWrapper<Employee>(json, s_deserializerOptionsPreserve));
-            Assert.Equal("$.$ref", ex.Path);
+            Assert.Equal("$", ex.Path);
```

Comment:
The changes in the JsonPath are due to the fact that reference metadata is passed to the ReferenceHandler at a later stage. This should have no noticeable impact on user experience, the error message indicates that it's due to a nonexisting reference.

Reply:
That discrepancy is intended. For example, if you deserialize a metadata property with invalid type, e.g.
```json
{ "$id" : false }
```
The error message will incorporate `$id` in the JsonPath. That's because the JSON violates System.Text.Json's metadata contract. On the other hand a JSON value such as
```json
{ "$id" : "badReferenceId", "$type" : "derivedClass", .... }
```
does not violate the metadata contract, but the string might get rejected by the reference resolver at a later stage by the DerivedClass converter. In this case, `$id` is not surfaced in the JsonPath (instead the existing error message clearly indicates that the `$id` value has been rejected by the resolver.

Reply:
> The error message will incorporate $id in the JsonPath. That's because the JSON violates System.Text.Json's metadata contract.

I agree with this, those errors will always occur at the moment a metadata property is read and it feels natural to display $id, ref or other metadata in the exception path.

> does not violate the metadata contract, but the string might get rejected by the reference resolver at a later stage by the DerivedClass converter.

As long as is consistent, it should be fine IMO. Will the path always point to the parent on all cases where there are non-contract errors on metadata?

Reply:
> Will the path always point to the parent on all cases where there are non-contract errors on metadata?

That's right.

Reply:
I might be missing something but if there is no reference resolver then `$ref` and `$id` are meaningless which means we should be showing `$ref` in the path